ibm_vpd_app.cpp

```diff
@@ -568,6 +568,43 @@ static void preAction(const nlohmann::json& json, const string& file)
             executePostFailAction(json, file);
         }
     }
+    else
+    {
+        // If the FRU is not there, clear the VINI/CCIN data.
+        // Enity manager probes for this keyword to look for this
+        // FRU, now if the data is persistent on BMC and FRU is
+        // removed this can lead to ambiguity. Hence clearing this
+        // Keyword if FRU is absent.
+        try
+        {
+            const auto& invPath =
+                json["frus"][file].at(0).value("inventoryPath", "");
+
+            if (!invPath.empty())
+            {
+                auto bus = sdbusplus::bus::new_default();
+                auto method = bus.new_method_call(
+                    INVENTORY_MANAGER_SERVICE, (pimPath + invPath).c_str(),
+                    "org.freedesktop.DBus.Properties", "Set");
+                method.append((std::string)ipzVpdInf + "VINI");
+                method.append("CC");
+                Binary ccData = {0x00, 0x02, 0x00, 0x00};
```

Comment:
Why don't we just use an empty vector (the PIM default)

Reply:
Fixed, In fact I made a call to PIM utility method itself to update the cache.